Supply Chain Workflow: SCITT Enhancement to Existing Services

SteveLasker opened the pull request:
Latest updates:
- More workflow feedback, with an additional placeholder doc for the detailed SCITT API workflow
- Split the single blob post to 3 separate files (what-is-supply-chain, distribution-with-oci-registries, distributing-with-oci-scitt

I'm suggesting:
- we remove the single blog post file
- merge this PR
- Start a new PR to merge in content from https://github.com/ietf-scitt/use-cases/blob/main/scitt-components.md

yogeshbdeshpande commented:
@SteveLasker Very nice work! Excellent start

rjb4standards commented:
Spot on Steve. During yesterdays DHS C-SCRM round table it became very clear that procurement officers are going to need automation to deal with the 1000's of vendors and 1000's of product and 1000's of self-attestation and 1000's of SBOM they need to collect and verify as trustworthy under OMB M-22-18. This will require automation; it cannot be done manually; https://energycentral.com/c/pip/advice-software-vendors-prepare-omb-m-22-18-requirements

rjb4standards commented:
Steve, just confirm; your reference to OCI pertains to this specification - correct?
https://github.com/opencontainers/distribution-spec/blob/main/spec.md#introduction

SteveLasker commented:
> Steve, just confirm; your reference to OCI pertains to this specification - correct? https://github.com/opencontainers/distribution-spec/blob/main/spec.md#introduction

@rjb4standards, yes, that's one of the references. It's hard to isolate just one reference as there are several OCI specs and details, but it's about storing artifacts and related artifacts in OCI Distribution based registries. This includes all the cloud providers (AWS, Azure, Google, Digital Ocean, ...) and several registry products and projects, including JFrog, Harbor and many others.

SteveLasker commented on _posts/2022-10-25-distributing-artifacts.md:
```diff
@@ -0,0 +1,94 @@
+
+### Supply Chain Integrity and Trust
+
+When you consume an artifact from a new location, how does the user know it originated from a trusted producer? Do you only purchase aspirin directly from Tylenol? Do you only acquire software directly from a vendor, or do you promote to a secured internal location within your control, just as you place the bottle of aspirin in medicine cabinet?
```
woops, good catch. I thought Aspirin was the "generic", but great catch. I'm going to tone down the "murder" reference as well :)

SteveLasker commented on _posts/2022-10-25-distributing-artifacts.md:
```diff
@@ -0,0 +1,94 @@
+
+How does the definition relate to a software supply chain?
+
+A supply chain spans multiple entities, requiring the promotion of artifacts across environments. In the physical world, this uses vehicles to transport (promote) between locations. In the digital world, a network is used to transport (promote) to different registries.
```
@roywill, are you referring to the use of registries? I'm using registries in the generic sense. Would it read easier if it said package managers as I'm referring to them as a group?
Companies, including Microsoft, import all the npm, debian, nuget, crate, container images, and even the open source they depend upon into their own environment. This was the promote to registries reference:

roywill replied:
Build systems pull via the appropriate package manager.  Whether this is backed by the registry is abstracted away.  Dev speak Nuget\NPM.

SteveLasker replied:
Ok, so you're reacting to the registry being a container specific thing, as the generic term for all package managers. I'll update.

SteveLasker commented:
I've updated the SCITT portion of the workflow, and incorporated the feedback.
I'm switching from DRAFT to ready for merge, with the assumption updates will come.

Preview at: https://stevelasker.github.io/scitt-web/distributing-with-oci

SteveLasker commented:
Latest updates:
- More workflow feedback, with an additional placeholder doc for the detailed SCITT API workflow
- Split the single blob post to 3 separate files (what-is-supply-chain, distribution-with-oci-registries, distributing-with-oci-scitt

I'm suggesting:
- we remove the single blog post file
- merge this PR
- Start a new PR to merge in content from https://github.com/ietf-scitt/use-cases/blob/main/scitt-components.md

roywill reviewed (approved):
Thanks Steve.  Captures the basics of progression from OCI towards a SCITT solution.
